### ISSUE TITLE
Unclear what to do about rtcp-mux/rtcp-rsize if first m= section is for data channels

### ISSUE THREAD

**Issue by juberti** [rtcweb-wg/jsep#528]
Say you have a session where only a data channel is created. SDP looks like

[session boilerplate]
a=group:BUNDLE d1
m=X application UDP/DTLS/SCTP webrtc-datachannel
a=mid:d1
...

Now, a media m= section is added (i.e. after first o/a). SDP becomes:

[session boilerplate]
a=group:BUNDLE d1 a1
m=X application UDP/DTLS/SCTP webrtc-datachannel
a=mid:d1
a=ice-ufrag:ABCD
...
m=X audio UDP/TLS/RTP/SAVPF 0
a=mid:a1
[no ice-ufrag or TRANSPORT attrs, since bundled]

Now, where does a=rtcp-mux go? It can't go into |d1| because the data section paragraph doesn't allow for it. And it can't go into |a1| because of the language that states

> [IDENTICAL/TRANSPORT] MUST appear only in "m=" sections which either have a unique address or which are associated with the bundle-tag. 

Thoughts on how to resolve?

**Comment by taylor-b** [comment]
Could generating a session description be split into two parts? First, generate media-level attributes for each "m=" section, then transport-level attributes for the BUNDLE group and each non-bundled "m=" section.

**Comment by juberti** [comment]
That would be my preference, i.e., jam it into the BUNDLE group m= section, as that's conceptually simplest, and easily handled here in JSEP.

Either way, I suspect this will require changes upstream in the bundle spec.

**Comment by fluffy** [comment]
So ... is it really not allowed in the d1 section. Unknown attributes are allowed in the d1 section, they are just ignored if they have no meaning for the data channel. Agree this sounds like something that will hit bundle.

**Comment by ekr** [comment]
Couldn't you do a=group:BUNDLE a1 d1

I would have thought this was prohibited, but is it really?
https://tools.ietf.org/html/draft-ietf-mmusic-sdp-bundle-negotiation-36#section-8.5.2

**Comment by taylor-b** [comment]
I think that's legal, but that approach doesn't work in the general case, where "m=" section A has TRANSPORT attributes that only it uses, and section B has TRANSPORT attributes that only it uses.

BUNDLE just needs to allow an "m=" section to contain attributes that it normally wouldn't contain if it wasn't bundled. (If this isn't already allowed).

**Comment by ekr** [comment]
@taylor-b: any action needed here?

**Comment by taylor-b** [comment]
I think what we want in this scenario is to put "a=rtcp-mux" and "a=rtcp-rsize" in the data channel section. The simplest way would be to say:

> If an RTP-based m= section is being bundled into the data m= section, add "a=rtcp-mux" and "a=rtcp-rsize" lines to the data m= section.

Which I guess could only happen in subsequent offers.

**Comment by juberti** [comment]
Let's go with what @taylor-b suggests, although this makes it a bundle problem.

**Comment by taylor-b** [comment]
What would you say is needed on the BUNDLE side? Just clarification that the offerer BUNDLE-tag m= section should contain the union of all TRANSPORT and IDENTICAL attributes used by the BUNDLE group?

**Comment by fluffy** [comment]
talked to chairs and advice from chairs on this and similar things that depend on mmusic outcomes. We put in JSEP now what our best guess is of how mmusic will end up and publish JSEP. At the same time we log a bug to check if we need to make any changes to line up with what mmusic decides. We can take changes to line up with mmusic as Last Call comments later on.

**Comment by fluffy** [comment]
+1 with Taylor proposal to put RTP related things in data m= section if it is first and good with making this be bundles problem

**Comment by ekr** [comment]
@juberti, @taylor-b: I assume that you think that on initial offers this cannot happen because data comes last?

**Comment by taylor-b** [comment]
@ekr Right. I can't think of a scenario where it wouldn't go last in an initial offer.

**Comment by juberti** [comment]
Agreed.

**Comment by ekr** [comment]
Oh, except that you aren't required to make the bundle group in order
So you can have
```
a=group:BUNDLE d1 a1
m=audio 10000 RTP/AVP 0 8 97
a=mid:a1
m=application 0 UDP/DTLS/SCTP webrtc-datachannel
a=mid:d1
```

**Comment by cdh4u** [comment]
I don't think the question whether you are allowed or not to place an RTP attribute in a non-RTP m= section is a pure BUNDLE issue. One needs to check what the specs defining those attributes say.

**Comment by fluffy** [comment]
every SDP spec has to accept (and ignore) lines it does not understand ...

**Comment by juberti** [comment]
I looked into the a=rtcp-mux and a=rtcp-rsize specs, and I didn't see anything that prohibited their inclusion into a non-media section.  

Generally, I think BUNDLE can define the rules for this sort of thing, without fear that it will somehow contradict some other spec.

**Comment by taylor-b** [comment]
@fluffy @juberti You may want to chime in on the current MMUSIC thread. I agree with you, but it seems to be going in the direction of "you can't do this".

**Comment by cdh4u** [comment]
>I looked into the a=rtcp-mux and a=rtcp-rsize specs, and I didn't see anything that prohibited their >inclusion into a non-media section.

It may not be explicitly stated, but since the semantics is about RTP and RTCP I think it is implicit. 

Having said that, I personally have no problem with allowing the attributes in non-RTP m- sections. I just want to verify that there is no text that has to be modified somewhere, and/or whether there is some text that has to be added somewhere :)

**Comment by juberti** [comment]
Resolved by #607
